src/python-common/ceph/deployment/translate.py

```diff
@@ -25,27 +25,38 @@ def __init__(self,
         self.osd_id_claims = osd_id_claims
 
     def run(self):
-        # type: () -> Optional[str]
+        # type: () -> List[str]
         """ Generate ceph-volume commands based on the DriveGroup filters """
         data_devices = [x.path for x in self.selection.data_devices()]
         db_devices = [x.path for x in self.selection.db_devices()]
         wal_devices = [x.path for x in self.selection.wal_devices()]
         journal_devices = [x.path for x in self.selection.journal_devices()]
 
         if not data_devices:
-            return None
+            return []
 
-        cmd = ""
+        cmds: List[str] = []
```

Comment:
I realize that the previous code returned a string, but I usually prefer command handling code to use lists/arrays/vectors (whatever you want to call 'em), rather than strings. IOW return List[List[str]] and instead of each individual command being a string its a List of args.  This is generally safer practice that avoids shell injection issues.  But I understand if that's too big a difference at this time.

Reply:
I'd like to leave it like this for now as I think changing it would actually require changing quite a bit (including every test around this function). I will mention however that the command generated here does just end up getting converted to a list of strings anyway before we actually make use of it https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/services/osd.py#L355.